### PR TITLE
Php8 and symfony5

### PR DESCRIPTION
**Pull request by GoranDimitrovski**

### CODE REVIEW

**Review by kudrmudr — APPROVED** [review]

File: tests/EventListener/SecurityListenerTest.php [hunk 1]
```diff
@@ -26,13 +29,25 @@ public function setViewParam()
...
 
+        $request = $this->getMockBuilder(Request::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $controller = $this->getMockBuilder(stdClass::class)
+            ->addMethods(['__invoke'])
+            ->getMock();
+
+        $requestType = HttpKernelInterface::MAIN_REQUEST;
```
Comment (pavbis) [inline tests/EventListener/SecurityListenerTest.php:46]: Where this constant comes from?
I can't find it in Symfony's interface
https://github.com/symfony/http-kernel/blob/5.2/HttpKernelInterface.php

File: composer.json [hunk 2]
```diff
@@ -3,23 +3,19 @@
...
+    "friendsofsymfony/rest-bundle": "^3.0",
+    "sensio/framework-extra-bundle": "^6.1"
   },
   "require-dev": {
-    "symfony/framework-bundle": "~3.4",
-    "symfony/validator": "^3.4",
-    "matthiasnoback/symfony-dependency-injection-test": "^2.0"
+    "matthiasnoback/symfony-dependency-injection-test": "^4.2",
+    "mockery/mockery": "^1.4",
+    "phpspec/prophecy-phpunit": "^2.0",
```
Comment (pavbis) [inline composer.json:14]: I would rather remove all prophecy calls in the tests and use native mocking stuff from phpunit as installing another dep to bring that "tests" running. There is a reason I guess why phpunit decided to remove prophecy.

**Review by pavbis — CHANGES_REQUESTED** [review]

File: composer.json (hunk 2 above)
Comment (GoranDimitrovski) [reply composer.json:15]: its from the version 5.3 https://github.com/symfony/http-kernel/blob/5.3/HttpKernelInterface.php

File: tests/EventListener/SecurityListenerTest.php (hunk 1 above)
Comment (GoranDimitrovski) [reply tests/EventListener/SecurityListenerTest.php:46]: its from the version 5.3 https://github.com/symfony/http-kernel/blob/5.3/HttpKernelInterface.php

File: composer.json (hunk 2 above)
Comment (GoranDimitrovski) [reply composer.json:14]: Maybe we can do this in a follow up ticket

File: tests/EventListener/SecurityListenerTest.php (hunk 1 above)
Comment (kudrmudr) [reply tests/EventListener/SecurityListenerTest.php:46]: but in require-dev you have symfony 5.2

Comment (GoranDimitrovski) [reply tests/EventListener/SecurityListenerTest.php:46]: @kudrmudr That is true but even if the required dev packages are set to 5.2
"symfony/framework-bundle": "5.2",
"symfony/validator": "5.2"
Their required packages are not required with a specific version. So thats how we end up at the end with symfony/http-kernel with version 5.3

**Review by pavbis — APPROVED** [review]

**Pull request merged**